File: premies/internal/grpc/card_prices.py
from gen.card_prices import card_prices_pb2, card_prices_pb2_grpc
from internal.service import card_prices


class CardPricesServiceServicer(card_prices_pb2_grpc.CardPricesServiceServicer):
    def UploadCardPricesData(self, request, context):
        try:
            result = card_prices.upload_card_prices(request.file_path)
            return card_prices_pb2.CardPricesUploadResponse(status=result)
        except Exception as e:
            return card_prices_pb2.CardPricesUploadResponse(status=f"Error: {str(e)}")

File: internal/controllers/automation/card_prices.go
package automation

import (
	"context"
	"github.com/gin-gonic/gin"
	"premiesPortal/internal/app/grpc/gen/card_prices"
	"premiesPortal/internal/clients/automation_premies/grpc"
	"premiesPortal/internal/controllers"
	"premiesPortal/pkg/errs"
)

func UploadCardPrices(c *gin.Context) {
	filePath := c.GetString(UploadedAutomationFilePath)

	if filePath == "" {
		controllers.HandleError(c, errs.ErrInvalidFilePath)
		return
	}

	client := grpc.GetClient()

	resp, err := client.UploadCardPricesData(context.Background(), &card_prices.CardPricesUploadRequest{FilePath: filePath})
	if err != nil {
		controllers.HandleError(c, err)
		return
	}

	c.JSON(resp.StatusCode, gin.H{
		"message": resp.Resp,
	})
}
